code/game/dna/dna_modifier.dm

```diff
@@ -371,29 +381,25 @@
 	return
 
 /obj/machinery/computer/scan_consolenew/ex_act(severity)
-
 	switch(severity)
 		if(1.0)
-			//SN src = null
 			qdel(src)
 			return
 		if(2.0)
 			if (prob(50))
-				//SN src = null
 				qdel(src)
 				return
-		else
-	return
 
 /obj/machinery/computer/scan_consolenew/blob_act()
-
 	if(prob(75))
 		qdel(src)
 
 /obj/machinery/computer/scan_consolenew/New()
 	..()
 	for(var/i=0;i<3;i++)
-		buffers[i+1]=new /datum/dna2/record
+		buffers[i+1] = new /datum/dna2/record
+	for(var/i=0;i<DNA_SE_LENGTH;i++)
```

Comment:
I struggled a lot trying to get it to work in javascript which does index at 0, I guess that rubbed off